sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -291,22 +360,24 @@ case class DataSource(
             throw new AnalysisException(s"Path does not exist: $qualified")
           }
           // Sufficient to check head of the globPath seq for non-glob scenario
-          if (checkFilesExist && !fs.exists(globPath.head)) {
+          // Don't need to check once again if files exist in streaming mode
+          if (!isStreaming && !fs.exists(globPath.head)) {
             throw new AnalysisException(s"Path does not exist: ${globPath.head}")
           }
           globPath
         }.toArray
 
-        // If they gave a schema, then we try and figure out the types of the partition columns
-        // from that schema.
-        val partitionSchema = userSpecifiedSchema.map { schema =>
-          StructType(
-            partitionColumns.map { c =>
-              // TODO: Case sensitivity.
-              schema
-                  .find(_.name.toLowerCase() == c.toLowerCase())
-                  .getOrElse(throw new AnalysisException(s"Invalid partition column '$c'"))
-            })
+        val (schema, inferredPartitionColumns) = getOrInferFileFormatSchema(format)
+        val dataSchema = if (isStreaming) {
+          schema
+        } else {
+          StructType(schema.dropRight(inferredPartitionColumns.length))
+        }
+
+        val partitionSchema = if (inferredPartitionColumns.isEmpty) {
```

Comment:
This is the one i dont understand. why should the behavior be different if this non-streaming source (HadoopFSRelation) is being create by user directly or by the file streaming source.